### PR TITLE
[#161] Added workflow runtime dependencies

### PR DESCRIPTION
**Pull request by kthoms**
- import package org.apache.commons.logging
- add com.ibm.icu as additional bundle
- remove XX:MaxPermSize arg from launch config

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties [hunk 1]
```diff
@@ -12,4 +12,7 @@ additional.bundles = org.eclipse.xtext.xbase,\
                      org.eclipse.xtext.xtext.generator,\
                      org.eclipse.emf.codegen.ecore,\
                      org.eclipse.emf.mwe.utils,\
-                     org.eclipse.emf.mwe2.launch
+                     org.eclipse.emf.mwe2.launch,\
+                     org.apache.log4j,\
+                     org.apache.commons.logging,\
+                     com.ibm.icu
```
Comment (LorenzoBettini) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: To be consistent with what the Xtext project wizard creates (and the first mwe2 run), this is the correct content:

```
additional.bundles = org.eclipse.xtext.xbase,\
                     org.eclipse.xtext.common.types,\
                     org.eclipse.xtext.xtext.generator,\
                     org.eclipse.emf.codegen.ecore,\
                     org.eclipse.emf.mwe.utils,\
                     org.eclipse.emf.mwe2.launch,\
                     org.eclipse.emf.mwe2.lib,\
                     org.objectweb.asm,\
                     org.apache.commons.logging,\
                     org.apache.log4j,\
                     com.ibm.icu
```

so you still miss some of them

Comment (kthoms) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: But why should there be additional dependencies which are already listed as required bundle? The bundles listed there should only complete the runtime classpath for the Xtext generator with bundles that aren't on the classpath by Require-Bundle or Import-Package.

The actual list of additional bundles that are sufficient to execute the generator are:
```
additional.bundles = org.eclipse.xtext.xtext.generator,\
                     org.eclipse.emf.codegen.ecore,\
                     org.eclipse.emf.mwe.utils,\
                     org.eclipse.emf.mwe2.launch,\
                     org.eclipse.emf.mwe2.lib,\
                     org.apache.commons.logging,\
                     org.apache.log4j,\
                     com.ibm.icu
```

So I think this is rather a problem of the wizard. It should not add the dependencies
- org.eclipse.xtext.xbase
- org.eclipse.xtext.common.types
- org.objectweb.asm
as additional bundle.

If you create a new project, remove those dependencies and run the Xtext generator, you should see that it succeeds and no compile error exists. (Actually I get a compile error in .tests, but this seems to be another issue that I'll raise now).

Do you agree that I should raise an issue for the wizard?

Comment (kthoms) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: The compile error I mentioned was a phantom one. It did not appear with a fresh project.

Comment (kthoms) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: Raised https://github.com/eclipse/xtext-core/issues/254

Comment (LorenzoBettini) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: see https://github.com/eclipse/xtext-core/issues/254#issuecomment-273139000 I think it's safer to leave them as they are

Comment (kthoms) [inline org.eclipse.xtext.xtext.ui.examples/projects/org.eclipse.xtext.example.domainmodel/build.properties]: OK, I'll align the build.properties with what the wizard would produce.

**Comment by kthoms** [comment]
@LorenzoBettini Changed as requested. Please have a look again.

**Pull request merged**